### PR TITLE
Implemented socket.io connection in place of APIs to eliminate pooling.

### PR DESCRIPTION
**Pull request by leandro-br-dev**
Implemented new feature using socket.io to replace constant polling by APIs, resolving issue #16. This clears console clutter, aids debugging, and reduces server overhead. All existing APIs and connections remain intact.

### CODE REVIEW

**Comment by darrassi1** [comment]
Is it compatible with duckduckgo api?

Because eventlet isn't

**Comment by leandro-br-dev** [comment]
> Is it compatible with duckduckgo api?
> 
> Because eventlet isn't

The implementation using Socket.io is not compatible with the DuckDuckGo API because they are different technologies with different purposes. Socket.io is a bidirectional communication protocol between client and server, while APIs like DuckDuckGo are used to request and receive data in a more traditional way, following the request-response pattern of the HTTP protocol. Therefore, Socket.io does not replace nor is compatible with APIs like DuckDuckGo, as they operate in different ways.

**Comment by darrassi1** [comment]
> > Is it compatible with duckduckgo api?
> > Because eventlet isn't
> 
> The implementation using Socket.io is not compatible with the DuckDuckGo API because they are different technologies with different purposes. Socket.io is a bidirectional communication protocol between client and server, while APIs like DuckDuckGo are used to request and receive data in a more traditional way, following the request-response pattern of the HTTP protocol. Therefore, Socket.io does not replace nor is compatible with APIs like DuckDuckGo, as they operate in different ways.

I mean when I use  duckduckgo_search library in the back-end
eventlet the communication protocol make the terminal of back-end stuck ,it's not showing Devika Is up and running . So I get rid of Duckduckgo_search and everything work again.
I wonder if I remove eventlet and use Socket.io with duckduckgo_search is they gonna work  in harmony together, or I will face a compatibility issue

**Comment by ARajgor** [comment]
#24

**Comment by franklinselva** [comment]
@leandro-br-dev Do you mind me fetching this work to my fork at some point?

**Comment by leandro-br-dev** [comment]
> I mean when I use duckduckgo_search library in the back-end eventlet the communication protocol make the terminal of back-end stuck ,it's not showing Devika Is up and running . So I get rid of Duckduckgo_search and everything work again. I wonder if I remove eventlet and use Socket.io with duckduckgo_search is they gonna work in harmony together, or I will face a compatibility issue

In the branch where I worked, DuckDuckGo Search wasn't implemented yet, so I can't confirm if it conflicts with Socket.io. However, since Socket.io and DuckDuckGo Search are different protocols, they shouldn't directly interfere with each other. You wouldn't be able to connect Socket.io directly to DuckDuckGo Search as they operate on different protocols. Socket.io deals with real-time communication, while DuckDuckGo Search is an API for search queries.
In other words, you can use Socket.io in conjunction with the Eventlet library, but you can't use Socket.io to connect directly to an API.

I hope that's clear, but if you have any further questions or doubts, feel free to ask.

**Comment by leandro-br-dev** [comment]
> @leandro-br-dev Do you mind me fetching this work to my fork at some point?

No worries, go ahead and fetch it!
